fixes.inc

```diff
@@ -14281,6 +14411,242 @@ native BAD_DestroyActor(actorid) = DestroyActor;
+ */
+
+#if _FIXES_SAMP && defined _ALS_GetVehicleComponentInSlot
+	#error _ALS_GetVehicleComponentInSlot defined
+#endif
+
+native BAD_GetVehicleComponentInSlot(vehicleid, slot) = GetVehicleComponentInSlot;
+
+#if FIX_GetVehicleComponentInSlot
+
+	stock FIXES_GetVehicleComponentType(componentid)
+	{
+		if(componentid == 1100 || componentid == 1115 || componentid == 1116 || componentid == 1123 || componentid == 1125)
+		{
+			return CARMODTYPE_FRONT_BULLBAR;
+		}
+		else if(componentid == 1109 || componentid == 1110)
+		{
+			return CARMODTYPE_REAR_BULLBAR;
+		}
+		return GetVehicleComponentType__(componentid);
+	}
+
+	stock FIXES_DestroyVehicle(vehicleid)
+	{
+		new const ret = DestroyVehicle__(vehicleid);
+		if(ret == 0) 
+		{
+			return ret;
+		}
+		#if !FIXES_Single		
+			setproperty(.name = FIXES_gscStereo, .value = 0);
+			setproperty(.name = FIXES_gscFrontBumper, .value = 0);
+			setproperty(.name = FIXES_gscRearBumper, .value = 0);
+			setproperty(.name = FIXES_gscFrontBullBar, .value = 0);
+			setproperty(.name = FIXES_gscRearBullbar, .value = 0);
+		#else
+			FIXES_gsStereo[vehicleid] = 0;
+			FIXES_gsFrontBumper[vehicleid] = 0;
+			FIXES_gsRearBumper[vehicleid] = 0;
+			FIXES_gsFrontBullBar[vehicleid] = 0;
+			FIXES_gsRearBullbar[vehicleid] = 0;
+		#endif
+		return ret;
+
+	}
+
+	stock FIXES_AddVehicleComponent(vehicleid, componentid)
+	{
+		new const ret = AddVehicleComponent__(vehicleid, componentid);
+		if(ret == 0) 
+		{
+			return ret;
+		}
+		switch(FIXES_GetVehicleComponentType(componentid))
+		{
+			#if !FIXES_Single
+			
+				case CARMODTYPE_STEREO:
+				{
+					setproperty(.name = FIXES_gscStereo, .value = componentid);
+				}
+				case CARMODTYPE_FRONT_BUMPER:
+				{
+					setproperty(.name = FIXES_gscFrontBumper, .value = componentid);
+				}
+				case CARMODTYPE_REAR_BUMPER:
+				{
+					setproperty(.name = FIXES_gscRearBumper, .value = componentid);
+				}
+				case CARMODTYPE_FRONT_BULLBAR:
+				{
+					setproperty(.name = FIXES_gscFrontBullBar, .value = componentid);
+				}
+				case CARMODTYPE_REAR_BULLBAR:
+				{
+					setproperty(.name = FIXES_gscRearBullbar, .value = componentid);
+				}
+			#else
+				case CARMODTYPE_STEREO:
+				{
+					FIXES_gsStereo[vehicleid] = componentid;
+				}
+				case CARMODTYPE_FRONT_BUMPER:
+				{
+					FIXES_gsFrontBumper[vehicleid] = componentid;
+				}
+				case CARMODTYPE_REAR_BUMPER:
+				{
+					FIXES_gsRearBumper[vehicleid] = componentid;
+				}
+				case CARMODTYPE_FRONT_BULLBAR:
+				{
+					FIXES_gsFrontBullBar[vehicleid] = componentid;
+				}
+				case CARMODTYPE_REAR_BULLBAR:
+				{
+					FIXES_gsRearBullbar[vehicleid] = componentid;
+				}
+			#endif
+		}
+		return ret;
+	}
+
+	stock FIXES_RemoveVehicleComponent(vehicleid, componentid)
+	{
+		new const ret = RemoveVehicleComponent__(vehicleid, componentid);
+		if(ret == 0) 
+		{
+			return ret;
+		}
+		switch(FIXES_GetVehicleComponentType(componentid))
+		{
+			#if !FIXES_Single
```

Comment:
`#if !/#else`.  It's just slightly confusing because that means the `#else` is `#if !!FIXES_Single`.